apps/studio/components/interfaces/Auth/ThirdPartyAuthForm/index.tsx

```diff
@@ -0,0 +1,160 @@
+    return null
+  }
+
+  if (isError) {
+    return (
+      <AlertError
+        error={error}
+        subject="Failed to retrieve auth configuration for Third Party Auth Integrations"
+      />
+    )
+  }
+
+  return (
+    <div className="pb-4">
+      <FormHeader
+        title="Third Party Auth"
+        description="Use third-party authentication (TPA) systems based on JWTs to access your project."
+        actions={
+          integrations.length !== 0 ? (
+            <AddIntegrationDropdown onSelectIntegrationType={setSelectedIntegration} />
+          ) : null
+        }
+      />
+
+      {isLoading && (
+        <div
+          className={cn(
+            'border rounded border-default px-20 py-16 flex flex-col items-center justify-center space-y-4'
+          )}
+        >
+          <Loader2 size={24} className="animate-spin" />
+        </div>
+      )}
+
+      {isSuccess ? (
+        integrations.length === 0 ? (
+          <div
+            className={cn(
+              'border rounded border-default px-20 py-16 flex flex-col items-center justify-center space-y-4'
+            )}
+          >
+            <p className="text-sm text-foreground-light">No integrations configured yet</p>
+            <AddIntegrationDropdown
+              buttonText="Add a new integration"
+              onSelectIntegrationType={setSelectedIntegration}
+            />
+          </div>
+        ) : (
+          <div className="-space-y-px">
+            {integrations.map((integration) => {
+              return (
+                <IntegrationCard
+                  key={integration.id}
+                  integration={integration}
+                  canUpdateConfig={canUpdateConfig}
+                  onSelect={() => {
+                    setSelectedIntegrationForDeletion(integration)
+                  }}
+                />
+              )
+            })}
+          </div>
+        )
+      ) : null}
+
+      <CreateFirebaseAuthIntegrationDialog
+        visible={selectedIntegration === 'firebase'}
+        onDelete={() => {}}
+        onClose={() => setSelectedIntegration(undefined)}
+      />
+
+      <CreateAwsCognitoAuthIntegrationDialog
+        visible={selectedIntegration === 'awsCognito'}
+        onDelete={() => {}}
+        onClose={() => setSelectedIntegration(undefined)}
+      />
+
+      <CreateAuth0IntegrationDialog
+        visible={selectedIntegration === 'auth0'}
+        onDelete={() => {}}
+        onClose={() => setSelectedIntegration(undefined)}
+      />
+
+      <ConfirmationModal
+        visible={!!selectedIntegrationForDeletion}
+        variant="destructive"
+        title="Confirm to delete"
+        confirmLabel="Delete"
+        confirmLabelLoading="Deleting"
+        onCancel={() => setSelectedIntegrationForDeletion(undefined)}
+        onConfirm={async () => {
+          if (!selectedIntegrationForDeletion) {
+            return
+          }
+          await deleteIntegration({
```

Comment:
i think the RQ invalidation here isn't working - the integration stays on the UI until i refresh the page

Reply:
Uploading Screen Recording 2024-07-20 at 01.04.17.mov…

Reply:
https://github.com/user-attachments/assets/d9b646ea-2e7f-4836-aae6-2121a07856cc

Reply:
This is really weird, works for me.

https://github.com/user-attachments/assets/d72cc449-231f-4c8d-8e50-ae36acdcf79a